File: py/prob_852.py
class Solution:
    def peakIndexInMountainArray(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        curr_max = 0
        curr_peak = 0
        for i in range(0, len(A)):
            if A[i] > curr_max:
                curr_max = A[i]
                curr_peak = i
        return curr_peak

File: cpp/prob_852.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& A) {
        int curr_max = 0;
        int curr_peak = 0;
        for (int i=0; i<A.size(); i++) {
            if (A[i] > curr_max) {
                curr_max = A[i];
                curr_peak = i;
            };
        };
        return curr_peak;
    };
};